File: Data Structures & Algorithms/majority-element/submission-1.java
class Solution {
    public int majorityElement(int[] nums) {

        Map<Integer, Integer> numCount = new HashMap<>();
        int res = 0, maxCount = 0;

        for(int num : nums) {
            numCount.put(num, numCount.getOrDefault(num, 0) + 1);
            
            if(numCount.get(num) > maxCount) {
                res = num;
                maxCount = numCount.get(num);
            }

            
        }

        return res;
        
    }
}

File: Data Structures & Algorithms/majority-element-ii/submission-1.cs
public class Solution {
    public List<int> MajorityElement(int[] nums) {
        
        var counts = new Dictionary<int, int>();
        var res = new List<int>();

        foreach (int num in nums) {
            if (!counts.ContainsKey(num)) {
                counts.Add(num, 0);
            }
            counts[num]++;
        }

        foreach(KeyValuePair<int, int> kvp in counts) {
            if (kvp.Value > nums.Length / 3) {
                res.Add(kvp.Key);
            }
        }

        return res;
        
    }
}